nectar/downloaders/threaded.py

```diff
@@ -160,6 +160,7 @@ def requests_kwargs_from_nectar_config(config):
                                          host=parsed_url.host, port=config.proxy_port)
             requests_kwargs['proxies'] = {'http': parsed_url.url, 'https': parsed_url.url}
 
+        requests_kwargs['stream'] = True # required for reading the download in chunks
```

Comment:
After digging into this, the nectar threaded downloader is setting the `self.session = True` in `__init__()` which means that:
```
session = Session()
session.stream = True
downloader = HTTPThreadedDownloader(...)
downloader.session = session
```
won't have the intended result.  As with the example, the streamer reassigns the session on the downloader provided by the plugin.

The fix here will ensure `stream=True` always regardless of how the session is configured (or created, passed, update after the downloader is constructed) and does not raise any thread safety concerns.

Here are the choices:
1. Accept this PR as is.
2. Accept this PR but request that `self.session.stream = True` be removed from `__init__()` so we have 1 place that *stream* is being managed (forced true).
3. Fix the stream as noted in the previous comment and reject this PR.

I'm leaning toward **(2)** because it restores nactar behavior before #54 was merged but in a thread safe way.